module/zfs/zfs_vnops.c

```diff
@@ -1387,6 +1387,17 @@ zfs_create(struct inode *dip, char *name, vattr_t *vap, int excl,
 			goto out;
 		}
 
+		/*
+		 * Check object quota
+		 */
+		if (zfs_fuid_overobjquota(zsb, B_FALSE, acl_ids.z_fuid) ||
```

Comment:
Why do this in a separate "if" statement, rather than adding this logic to  `zfs_acl_ids_overquota()`?